### ISSUE TITLE
"OCP\\AppFramework\\QueryException" when opening /settings/admin/additional

### ISSUE THREAD

**Issue by SimJoSt** [pawelrojek/nextcloud-drawio#63]
### Steps to reproduce
1. instal draw.io app
2. open "additional settings" (/settings/admin/additional)

### Expected behaviour
No errors in logs.

### Actual behaviour
Errors in log:
```
    "Exception": "OCP\\AppFramework\\QueryException",
    "Message": "Could not resolve OCA\\Drawio\\Settings\\AdminSettings! Class OCA\\Drawio\\Settings\\AdminSettings does not exist",
```

## Server configuration detail

**Operating system:** Linux 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64

**Webserver:** Caddy/0.11.1 (fpm-fcgi)

**Database:** mariadb 10.1.37

**PHP version:** 

7.2.12
Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, readline, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, apcu, exif, gd, intl, ldap, memcached, pcntl, pdo_mysql, pdo_pgsql, redis, sodium, zip, Zend OPcache

**Nextcloud version:** 14.0.4 - 14.0.4.2

**Updated from an older Nextcloud/ownCloud or fresh install:** updated

**Where did you install Nextcloud from:** Docker

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.0.1
 - activity: 2.7.0
 - admin_audit: 1.4.0
 - announcementcenter: 3.3.1
 - apporder: 0.6.0
 - audioplayer: 2.4.1
 - bruteforcesettings: 1.2.0
 - calendar: 1.6.4
 - camerarawpreviews: 0.6.1
 - checksum: 0.4.1
 - circles: 0.15.2
 - cloud_federation_api: 0.0.1
 - comments: 1.4.0
 - contacts: 2.1.7
 - dav: 1.6.0
 - deck: 0.5.0
 - drawio: 0.9.1
 - dropit: 0.1.3
 - external: 3.1.0
 - federatedfilesharing: 1.4.0
 - federation: 1.4.0
 - files: 1.9.0
 - files_downloadactivity: 1.3.0
 - files_external: 1.5.0
 - files_linkeditor: 1.0.7
 - files_markdown: 2.0.5
 - files_pdfviewer: 1.3.2
 - files_retention: 1.3.0
 - files_rightclick: 0.8.4
 - files_sharing: 1.6.2
 - files_texteditor: 2.6.0
 - files_trashbin: 1.4.1
 - files_versions: 1.7.1
 - files_videoplayer: 1.3.0
 - firstrunwizard: 2.3.0
 - flowupload: 0.0.8
 - gallery: 18.1.0
 - groupfolders: 1.3.3
 - impersonate: 1.1.0
 - issuetemplate: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.2.0
 - mail: 0.11.0
 - metadata: 0.8.0
 - mindmaps: 0.1.0
 - music: 0.9.2
 - nextcloud_announcements: 1.3.0
 - notes: 2.5.0
 - notifications: 2.2.1
 - oauth2: 1.2.1
 - onlyoffice: 2.0.4
 - ownpad: 0.6.8
 - password_policy: 1.4.0
 - polls: 0.8.3
 - previewgenerator: 2.0.0
 - provisioning_api: 1.4.0
 - quota_warning: 1.3.0
 - ransomware_protection: 1.2.0
 - richdocuments: 3.0.5
 - serverinfo: 1.4.0
 - sharebymail: 1.4.0
 - socialsharing_email: 1.0.4
 - spreed: 4.0.1
 - systemtags: 1.4.0
 - tasks: 0.9.8
 - theming: 1.5.0
 - twofactor_backupcodes: 1.3.1
 - twofactor_gateway: 0.11.0
 - twofactor_totp: 1.5.0
 - twofactor_u2f: 1.6.1
 - twofactor_yubikey: 0.4.0
 - updatenotification: 1.4.1
 - workflowengine: 1.4.0
Disabled:
 - encryption
 - files_accesscontrol
 - files_automatedtagging
 - files_reader
 - keeweb
 - support
 - survey_client
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "14.0.4.2",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "forcessl": true,
    "logtimezone": "Europe\/Berlin",
    "log_rotate_size": 104857600,
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtpauthtype": "LOGIN",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "465",
    "mail_smtpsecure": "ssl",
    "htaccess.RewriteBase": "\/",
    "htaccess.IgnoreFrontController": true,
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "skeletondirectory": "",
    "auth.bruteforce.protection.enabled": true,
    "check_for_working_webdav": true,
    "maintenance": false,
    "loglevel": 0,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "apps_paths": [
        {
            "path": "\/var\/www\/html\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/var\/www\/html\/custom_apps",
            "url": "\/custom_apps",
            "writable": true
        }
    ],
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "updater.release.channel": "stable",
    "preview_max_x": 400,
    "preview_max_y": 200
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** no

## Client configuration

**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36

**Operating system:** Windows 10 64bit

## Logs

<details><summary>Browser log</summary>

```

```
</details>

<details><summary>Nextcloud log</summary>

```
{
  "reqId": "FAZsB85dPJFaf0vLa2dW",
  "level": 1,
  "time": "2018-11-28T21:11:02+01:00",
  "remoteAddr": "***REMOVED SENSITIVE VALUE***",
  "user": "SimJoSt",
  "app": "no app in context",
  "method": "GET",
  "url": "\/settings\/admin\/additional",
  "message": {
    "Exception": "OCP\\AppFramework\\QueryException",
    "Message": "Could not resolve OCA\\Drawio\\Settings\\AdminSettings! Class OCA\\Drawio\\Settings\\AdminSettings does not exist",
    "Code": 0,
    "Trace": [{
      "file": "\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php",
      "line": 119,
      "function": "resolve",
      "class": "OC\\AppFramework\\Utility\\SimpleContainer",
      "type": "->",
      "args": ["OCA\\Drawio\\Settings\\AdminSettings"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/ServerContainer.php",
      "line": 132,
      "function": "query",
      "class": "OC\\AppFramework\\Utility\\SimpleContainer",
      "type": "->",
      "args": ["OCA\\Drawio\\Settings\\AdminSettings"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/Settings\/Manager.php",
      "line": 216,
      "function": "query",
      "class": "OC\\ServerContainer",
      "type": "->",
      "args": ["OCA\\Drawio\\Settings\\AdminSettings"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/Settings\/Manager.php",
      "line": 342,
      "function": "getSettings",
      "class": "OC\\Settings\\Manager",
      "type": "->",
      "args": ["admin", "additional"]
    }, {
      "file": "\/var\/www\/html\/settings\/Controller\/AdminSettingsController.php",
      "line": 73,
      "function": "getAdminSettings",
      "class": "OC\\Settings\\Manager",
      "type": "->",
      "args": ["additional"]
    }, {
      "file": "\/var\/www\/html\/settings\/Controller\/CommonSettingsTrait.php",
      "line": 126,
      "function": "getSettings",
      "class": "OC\\Settings\\Controller\\AdminSettingsController",
      "type": "->",
      "args": ["additional"]
    }, {
      "file": "\/var\/www\/html\/settings\/Controller\/AdminSettingsController.php",
      "line": 65,
      "function": "getIndexResponse",
      "class": "OC\\Settings\\Controller\\AdminSettingsController",
      "type": "->",
      "args": ["admin", "additional"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
      "line": 166,
      "function": "index",
      "class": "OC\\Settings\\Controller\\AdminSettingsController",
      "type": "->",
      "args": ["additional"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
      "line": 99,
      "function": "executeController",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->",
      "args": [{
        "__class__": "OC\\Settings\\Controller\\AdminSettingsController"
      }, "index"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/AppFramework\/App.php",
      "line": 118,
      "function": "dispatch",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->",
      "args": [{
        "__class__": "OC\\Settings\\Controller\\AdminSettingsController"
      }, "index"]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php",
      "line": 47,
      "function": "main",
      "class": "OC\\AppFramework\\App",
      "type": "::",
      "args": ["OC\\Settings\\Controller\\AdminSettingsController", "index", {
        "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
      }, {
        "section": "additional",
        "_route": "settings.AdminSettings.index"
      }]
    }, {
      "function": "__invoke",
      "class": "OC\\AppFramework\\Routing\\RouteActionHandler",
      "type": "->",
      "args": [{
        "section": "additional",
        "_route": "settings.AdminSettings.index"
      }]
    }, {
      "file": "\/var\/www\/html\/lib\/private\/Route\/Router.php",
      "line": 297,
      "function": "call_user_func",
      "args": [{
        "__class__": "OC\\AppFramework\\Routing\\RouteActionHandler"
      }, {
        "section": "additional",
        "_route": "settings.AdminSettings.index"
      }]
    }, {
      "file": "\/var\/www\/html\/lib\/base.php",
      "line": 987,
      "function": "match",
      "class": "OC\\Route\\Router",
      "type": "->",
      "args": ["\/settings\/admin\/additional"]
    }, {
      "file": "\/var\/www\/html\/index.php",
      "line": 42,
      "function": "handleRequest",
      "class": "OC",
      "type": "::",
      "args": []
    }],
    "File": "\/var\/www\/html\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php",
    "Line": 104,
    "CustomMessage": "--"
  },
  "userAgent": "Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/70.0.3538.110 Safari\/537.36",
  "version": "14.0.4.2"
}
```
</details>

<details><summary>Browser log</summary>

</details>

**Comment by janLo** [comment]
This is due to a completely broken fix for #4 in b74bf38daa15646fe5221a7bae910d562c327439. This "fixes" the doble appearance of the drawio settings by giving a non-existing class in the appinfo/info.xml.

The reason for the double appearance was that the config section is registered twice. Once in the info.xml and once in the app.php (https://github.com/pawelrojek/nextcloud-drawio/blob/master/drawio/appinfo/app.php#L16). The solution is to remove one of the registration options, not to make one causing an error by giving an invalid argument. 

I have to admit, that the whole settings-registration-process for nextcloud is a bit strange and changes seems to only have an effect after the server restart - so I can see why this has happened.

To move forward: the registration in app.php should be removed as this function is deprecated since NC14 (https://docs.nextcloud.com/server/stable/developer_manual/api/OCP/App.html?highlight=registeradmin) and the entry in info.xml should be corrected.

I'll prepare a PR for this.

**Comment by SimJoSt** [comment]
@janLo I can confirm the issue has been resolved.
